source/JsMaterialX/JsMaterialXCore/JsDocument.cpp

```diff
@@ -17,14 +18,16 @@ extern "C"
         ems::class_<mx::Document, ems::base<mx::GraphElement>>("Document")
             .smart_ptr_constructor("Document", &std::make_shared<mx::Document, mx::ElementPtr, const std::string &>)
             .smart_ptr<std::shared_ptr<const mx::Document>>("Document")
+             // At the moment only the Document type is used. Once more types are added this binding needs to be adapted accordingly.
+            .class_function("createDocument", &mx::Document::createDocument<mx::Document>)
             .function("initialize", &mx::Document::initialize)
             .function("copy", &mx::Document::copy)
             .function("importLibrary", &mx::Document::importLibrary)
             .function("getReferencedSourceUris", ems::optional_override([](mx::Document &self) {
-                          mx::StringSet referenced = self.mx::Document::getReferencedSourceUris();
-                          int size = referenced.size();
-                          return arrayToVec((std::string *)&referenced, size);
-                      }))
+                mx::StringSet referenced = self.mx::Document::getReferencedSourceUris();
+                int size = referenced.size();
+                return arrayToVec((std::string *)&referenced, size);
```

Comment:
I wonder if we could save a copy by creating a `val::array` instead of a vector here, similar to how we do it in the `getVersionIntegers` wrapper? This is also a candidate for explicit testing.